addons/compact-menu-bar/userscript.js

```diff
@@ -0,0 +1,18 @@
+export default async function ({ addon }) {
+  let span;
+  let li;
+  if (await addon.auth.fetchIsLoggedIn()) {
+    while (true) {
+      // Remove username from bar
+      const username = await addon.auth.fetchUsername();
+      const dropdown = await addon.tab.waitForElement(".account-nav_profile-name_2oRiV");
+      span = dropdown;
+      span?.remove();
```

Comment:
> The username hides appropriately from the menu bar, but it is shown in the account menu only one time, when clicking outside and re-opening it the username disappears. Any ideas on how I could fix this?

The problem is the script is waiting for this element to be recreated before moving on to the next one which only happens when re-entering the editor.

We prefer using CSS over JavaScript when possible, so the best solution is to move this to the userstyle with `display: none;` instead.

Reply:
This didn't fix the problem. Apparently, Scratch regenerates the CSS every time you switch the page.
What I don't understand is why the label is shown properly on the website, but not in the editor.